### PR TITLE
Snap 1325

### PR DESCRIPTION
**Pull request by BhattChandresh**
## Changes proposed in this pull request

- Added the test cases in Java for UDF1 to UDF22 using org.apache.spark.sql.api.java.* [* -> UDF1 to UDF22]
- Added the test cases in Scala for UDF1 to UDF22 using org.apache.spark.sql.api.java.* [* -> UDF1 to UDF22]
- Added the validation using assert statements.

## Patch testing
- Verified the changes by running udfs.bt.

## ReleaseNotes.txt changes
None

## Other PRs 
None

### CODE REVIEW

**Comment by sumwale** [comment]
One small comment: if possible its better to create udf.jar using a gradle task dynamically and use that instead of checking in udf.jar

**Comment by BhattChandresh** [comment]
Actually, that was the idea. Dynamically create the udf.jar with gradle task but do other priorities currently, deploy the udf.jar in tests. Will create the gradle task for it.

**Comment by PradeepSurale** [comment]
Can one of the admins verify this patch?

File: dtests/src/test/java/io/snappydata/hydra/udfs/JavaUDF3.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    public String call(WrappedArray<String> wstr1, WrappedArray<String> wstr2, WrappedArray<String> wstr3) throws Exception {
+        String[] arr1 = new String[wstr1.length()];
+        String[] arr2 = new String[wstr2.length()];
+        String[] arr3 = new String[wstr3.length()];
+        String outputStr = "";
+
+        wstr1.copyToArray(arr1);
+        wstr2.copyToArray(arr2);
+        wstr3.copyToArray(arr3);
+
```
Comment (sonalsagarwal) [inline dtests/src/test/java/io/snappydata/hydra/udfs/JavaUDF3.java]: Optimize the code for a converting string to upper case.

File: dtests/src/test/java/io/snappydata/hydra/udfs/JavaUDF3.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package io.snappydata.hydra.udfs;
+
+import org.apache.spark.sql.api.java.UDF3;
+import scala.collection.mutable.WrappedArray;
+
+public class JavaUDF3 implements UDF3<WrappedArray<String>, WrappedArray<String>, WrappedArray<String>, String> {
```
Comment (sonalsagarwal) [inline dtests/src/test/java/io/snappydata/hydra/udfs/JavaUDF3.java]: Please add descriptions for the functions.

File: dtests/src/test/java/io/snappydata/hydra/udfs/udf.conf [hunk 3]
```diff
@@ -0,0 +1,28 @@
+hydra.Prms-testRequirement = "Test for org.apache.spark.sql.api.java intefaces (UDF1 to UDF22) in Java/Scala";
+hydra.Prms-testDescription = "This is a good test case. This test create,execute and drop the functions in Java/Scala by using snappy methods : CREATE FUNCTION";
+
+INCLUDE $JTESTS/io/snappydata/hydra/northwind/startEmbeddedModeCluster.conf;
+
+//TASK taskClass =  io.snappydata.hydra.udfs.UDFInterface  taskMethod = HydraTask_JavaUDFS
+//                                 io.snappydata.hydra.cluster.SnappyPrms-userAppJar = ${udfJarPath}
```
Comment (sonalsagarwal) [inline dtests/src/test/java/io/snappydata/hydra/udfs/udf.conf]: Please resolve the commented tasks.

File: dtests/src/test/java/io/snappydata/hydra/udfs/JavaUDF3.java (hunk 2 above)
Comment (BhattChandresh) [reply dtests/src/test/java/io/snappydata/hydra/udfs/JavaUDF3.java]: Done.

File: dtests/src/test/java/io/snappydata/hydra/udfs/udf.conf (hunk 3 above)
Comment (BhattChandresh) [reply dtests/src/test/java/io/snappydata/hydra/udfs/udf.conf]: Done.

File: dtests/src/test/java/io/snappydata/hydra/udfs/JavaUDF3.java (hunk 1 above)
Comment (BhattChandresh) [reply dtests/src/test/java/io/snappydata/hydra/udfs/JavaUDF3.java]: Done.

**Pull request merged**